File: Selenium/Project/Activity6.py
# Import required packages from selenium
import time

import pytest
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait


class TestClass():

    def test_verify_header_text(self, my_fixture):

        # Initialise Driver
        driver = my_fixture
        url = 'https://alchemy.hguy.co/jobs'
        driver.get(url)

        # Initialise Wait for 10 secs
        wait = WebDriverWait(driver, 10)

        # Find and Click the Jobs Link
        jobs_link = driver.find_element(By.ID, "menu-item-24")
        jobs_link.click()

        # Search for Banking Jobs
        searchJob = driver.find_element(By.XPATH, "//input[@id='search_keywords']")
        searchButton = driver.find_element(By.XPATH, "//input[@type='submit']")
        searchJob.send_keys("Banking")
        searchButton.click()

        # Wait for the Visibility of the Banking Job and click on it
        wait.until(expected_conditions.visibility_of_element_located((By.XPATH, "//div[@class='job_listings']/ul/li[1]/a")))
        jobName = driver.find_element(By.XPATH, "//div[@class='job_listings']/ul/li[1]/a")
        jobName.click()

        # Apply for the Job
        applyButton = driver.find_element(By.XPATH, "//input[@class='application_button button']")
        applyButton.click()

        # Print the Confirmation email and Print it
        confirmMsg = driver.find_element(By.XPATH, "//a[@class='job_application_email']")
        email = confirmMsg.text
        print("Email Id is : " + email)

File: Selenium/Project/Activity6.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.junit.jupiter.api.Assertions;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.time.Duration;

public class Activity6 {
    WebDriver driver;
    WebDriverWait wait;

    @BeforeClass
    public void setup() {
        WebDriverManager.firefoxdriver().setup();
        driver = new FirefoxDriver();
        driver.get("https://alchemy.hguy.co/jobs");
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.manage().timeouts().implicitlyWait(Duration.ofMillis(5000));
    }

    @Test
    public void applyJobs() throws InterruptedException {

        WebElement jobsLink = driver.findElement(By.id("menu-item-24"));

        // Click the Job Link
        jobsLink.click();

        WebElement searchJob = driver.findElement(By.xpath("//input[@id='search_keywords']"));
        searchJob.sendKeys("Banking");

        WebElement searchButton = driver.findElement(By.xpath("//input[@type='submit']"));
        searchButton.click();

        WebElement jobName = driver.findElement(By.xpath("//div[@class='job_listings']/ul/li[1]/a"));
        jobName.click();

        WebElement applyButton = driver.findElement(By.xpath("//input[@class='application_button button']"));
        applyButton.click();

        WebElement confirmMsg = driver.findElement(By.xpath("//a[@class='job_application_email']"));
        String email = confirmMsg.getText();

        System.out.println("Email Id is : " + email);
    }

    @AfterClass
    public void cleanup() {
        driver.quit();
    }
}
